skrf/calibration/deembedding.py

```diff
@@ -1263,40 +1272,43 @@ def split2xthru(self, s2xthru):
             #     # mhuser : is it a problem with complex value cast to real here ?
             #     e10[i] = k * np.sqrt(s12r[i] * (1 - e111[i] * e112[i]))
             
-            # elegant way to avoid 180° phase jumps in case of phase noise
             # calc e01 and e10
-            e10e01 = 0.5 * (s12r + s21r) * (1 - e111 * e112)
-            e01 = np.sqrt(e10e01)
-            e10 = zeros(n, dtype = complex)
+            # avoid 180° phase jumps in case of phase noise
+            e01 = np.sqrt(s21r * (1 - e111 * e112))
             for i in range(n):
                 if i > 0:
                     if npy.abs(-e01[i] - e01[i-1]) < npy.abs(e01[i] - e01[i-1]):
-                        e01[i] *= -1
-                e10[i] = e10e01[i] / e01[i]
+                        e01[i] = - e01[i]
+            e10 = np.sqrt(s12r * (1 - e111 * e112))
+            for i in range(n):
+                if i > 0:
+                    if npy.abs(-e10[i] - e10[i-1]) < npy.abs(e10[i] - e10[i-1]):
+                        e10[i] = - e10[i]

```

Comment:
Is this functionality the same as `np.unwrap`?
https://numpy.org/doc/stable/reference/generated/numpy.unwrap.html

If yes, function would be much faster probably

Reply:
Sorry, did not recognice that you merged already :)

Reply:
No problem :)
It looks like unwrap but is not strictly similar: instead of adding `2*k*pi` to the value, it changes the angle of the complex with the value `pi`. This is because the square root solution can be positive or negative, so it takes the value that avoids a phase jump.